src/SFA.DAS.Tools.Servicebus.Support.Infrastructure/Services/CosmosDbContext.cs

```diff
@@ -226,5 +228,54 @@ private string AddPaging(string sqlQuery, SearchProperties searchProperties)
 
             return sb.ToString();
         }
+
+        public async Task<UserSession> UpsertUserSessionAsync(UserSession userSession)
+        {
+            var database = await _client.CreateDatabaseIfNotExistsAsync(_databaseName);
+            var container = await CreateContainer(database);
+            return await container.UpsertItemAsync(userSession);
+        }
+
+        public async Task<UserSession> GetUserSessionAsync(string userId)
+        {
+            var sqlQuery = $"SELECT * FROM c WHERE c.userId = '{userId}' and c.type = 'session'";
+
+            var database = await _client.CreateDatabaseIfNotExistsAsync(_databaseName);
+            var container = await CreateContainer(database);
+
+            var queryDefinition = new QueryDefinition(sqlQuery);
+            var queryFeedIterator = container.GetItemQueryIterator<UserSession>(queryDefinition);
+            var currentResults = await queryFeedIterator.ReadNextAsync();
+
+            return currentResults.FirstOrDefault();
+        }
+
+        public async Task DeleteUserSessionAsync(string id, string userId)
+        {
+            Database database = await _client.CreateDatabaseIfNotExistsAsync(_databaseName);
+            var container = await CreateContainer(database);
+
+            await container.DeleteItemAsync<UserSession>(id, new PartitionKey(userId));
+        }
+
+        public async Task<IEnumerable<UserSession>> GetExpiredUserSessions()
+        {            
+            var database = await _client.CreateDatabaseIfNotExistsAsync(_databaseName);
+            var container = await CreateContainer(database);
+
+            var queryFeedIterator = container.GetItemLinqQueryable<UserSession>().Where(s => s.ExpiryDateUtc < DateTime.UtcNow).ToFeedIterator();            
+
+            var userSessions = new List<UserSession>();
+
+            while (queryFeedIterator.HasMoreResults)
+            {
+                var currentResults = await queryFeedIterator.ReadNextAsync();
+
+                var newSessions = currentResults.ToList();
+                userSessions.AddRange(newSessions);
+            }
+
+            return userSessions;            
+        }
```

Comment:
Would it be worth separating these methods out into a specific Context on their own?

Reply:
@DannyGB refactored